app/models/katello/repository.rb

```diff
@@ -170,13 +170,13 @@ def self.in_environments_products(env_ids, product_ids)
 
     def other_repos_with_same_product_and_content
       list = Repository.in_product(Product.find(self.product.id)).where(:content_id => self.content_id).all
-      list.destroy(self)
+      list.destroy(self) if list.any?
       list
     end
 
     def other_repos_with_same_content
       list = Repository.where(:content_id => self.content_id).all
-      list.destroy(self)
+      list.destroy(self) if list.any?
```

Comment:
So when i looked at these methods, it made no sense that repos were being destroyed.  Looking at the history, this method was never meant to destroy the repo, it was meant to remove itself from the array.  

This commit:  https://github.com/Katello/katello/commit/72352008b1bd1ca3d4bdaee77b56650ec3bbe940#diff-7bfb4ceeca67af50fe464d5451281e40L170

made the change from:

   list.delete(self)  #aka remove self from the array
to
list.destroy(self)  #not the same thing

you might just add a `.where("#{self.class.table_name}.id != #{self.id}")`